### ISSUE TITLE
In KITTI test, the generated point cloud is not base on input

### ISSUE THREAD

**Issue by romantic-black** [yuxumin/PoinTr#142]
![image](https://github.com/yuxumin/PoinTr/assets/63280964/8e33107b-e4b0-4cba-8f55-fd4f5dea3244)
![image](https://github.com/yuxumin/PoinTr/assets/63280964/53a97a71-4811-4c55-bc3e-af7a2cc24748)

**Comment by kaali-billi** [comment]
The problem is the data manipulation during training and evaluation. The transformer works on normalized input and GT, but the normalizing factor and the centroid assumed are not the same as the GT scaling factor and the GT centroid. That is why there is error on reconstruction position and side. They have not mentioned this anywhere in the publication or the code. Took me 4 months to finally understand what was going wrong. I have started a new training cycle with changes to the input data to equalize the scaling and centroid. 

![pc_norm_gt](https://github.com/yuxumin/PoinTr/assets/61189809/d6910f64-b4f2-4fc0-88a0-74aa6efb7f0b)
![iss_correc](https://github.com/yuxumin/PoinTr/assets/61189809/2d007749-512d-4e4d-b327-70f8e7efd073)

**Comment by kaali-billi** [comment]
Extract the centroid(cen_par) and scaling factor(m_par) and use these to normalize the GT on every instance using pc_norm() and pc_norm_gt(). @yuxumin I believe you need to address this issue otherwise your publication is not completely correct.

**Comment by fabricecarles** [comment]
yes @kaali-billi the current implementation is not correct and I also make some changes as suggested in my issue #130. I found that results are far better in case of real world scenario (i.e when you make inference without any ground truth)

**Comment by MaSiTou** [comment]
@kaali-billi According to your code, should I first extract the partial points cloud and then normalize it with ground_truth
